src/evaluators/state/PHAL_LoadStateField_Def.hpp

```diff
@@ -40,6 +40,22 @@ void LoadStateFieldBase<EvalT, Traits, ScalarType>::postRegistrationSetup(typena
   if (d.memoizer_active()) memoizer.enable_memoizer();
 }
 
+// **********************************************************************
+template<typename EvalT, typename Traits, typename ScalarType>
+void LoadStateFieldBase<EvalT, Traits, ScalarType>::preEvaluate(typename Traits::EvalData workset)
+{
+  if (memoizer.have_saved_data(workset,this->evaluatedFields())) return;
```

Comment:
If this get's called in `preEvaluate` with memoization on, `evaluateFields` will get completely skipped when using a single workset since `_prevWorksetIndex ` gets set to zero here. It looks like the code is the same so I think that's okay, just wanted to point that out.

Reply:
I'm reverting this change.